core/trino-spi/src/main/java/io/trino/spi/PageBuilder.java

```diff
@@ -93,7 +93,7 @@ public void reset()
 
         declaredPositions = 0;
 
-        for (int i = 0; i < types.size(); i++) {
```

Comment:
why?

Reply:
Because the loop conditional assumes `types.size() == blockBuilders.length` (enforced in the constructor) but the JIT bounds check elimination would prefer that the termination condition refer the the array length of the array being traversed.